### ISSUE TITLE
Colorized traceback with RunServerPlus

### ISSUE THREAD

**Issue by dufferzafar** [django-extensions/django-extensions#1512]
There are 3rd party libraries available that colorize the python tracebacks to make them a more readable. Like, https://github.com/nir0s/backtrace, https://github.com/staticshock/colored-traceback.py

I was wondering how can I integrate these tools with RunServerPlus/werkzeug? Just importing them and setting the right hooks in manage.py doesn't really work.

**Comment by trbs** [comment]
That would be really cool :-)

Guess something like this could also work site wide (not specifically for `runserverplus`) but for any exception in the running Python VM.

**Comment by dufferzafar** [comment]
Yeah, it doesn't have to be `runserverplus` specific. I'd be up to take this, just need some guidance on how to approach it, am not really familiar with the internals of either werkzeug / or django run_server.

**Comment by trbs** [comment]
I did a small test with adding the `def ready(self)` method to Django Extension's AppConfig.
This seems to work :-) 

```
class DjangoExtensionsConfig(AppConfig):
    ...
    def ready(self):
        import colored_traceback
        colored_traceback.add_hook()
```

```./venv/bin/python manage.py shell_plus -c test```
<img width="729" alt="image" src="https://user-images.githubusercontent.com/65552/79861629-fa11d700-83d4-11ea-87fb-626ec1efc6cc.png">

Think this or a similar approach might work great.
It will (of course) install the hook for everything running inside the current project, but that seems logical to me.

From the top of my head we would further need:
 * list of 3rd party libraries which to support
 * add some settings to control this (enable, disable, set which hook/package to use, etc)
 * add documentation on how to use, install package, default behavior, etc.
 * some tests

* https://docs.djangoproject.com/en/3.0/ref/applications/#django.apps.AppConfig.ready

**Comment by dufferzafar** [comment]
Will it work with Werkzeug as well?

**Comment by trbs** [comment]
Maybe/Probably not... 
In my test it does not work for exceptions resulting from web requests in the forked process / using the custom code for printing stack traces.

See:
- https://github.com/pallets/werkzeug/issues/1753
- https://github.com/pallets/werkzeug/issues/1786

There might be some hope there :-)
